photutils/isophote/tests/test_isophote.py

```diff
@@ -97,11 +99,13 @@ def test_m51(self):
         assert iso.tflux_c <= 1.38e6
         assert iso.tflux_c >= 1.36e6
 
-        # deviations from perfect ellipticity
-        assert abs(iso.a3) <= 0.05
-        assert abs(iso.b3) <= 0.05
-        assert abs(iso.a4) <= 0.05
-        assert abs(iso.b4) <= 0.05
+        # deviations from perfect ellipticity. Note
+        # that sometimes a None covariance can be
+        # generated by scipy.optimize.leastsq
+        assert iso.a3 is None or abs(iso.a3) <= 0.05
+        assert iso.a3 is None or abs(iso.b3) <= 0.05
+        assert iso.a3 is None or abs(iso.a4) <= 0.05
+        assert iso.a3 is None or abs(iso.b4) <= 0.05
```

Comment:
Here too.

Reply:
Indeed, it's not the correct way to test for the None condition. But since the 4 parameters are all tied together, meaning, either none of them, or all of them, can have the value None, the test would work anyway. But would look funny to someone wanting to understand the logic.

Reply:
The next commit fixes it.

Reply:
Thanks, @ibusko.  Actually, this `Travis` test failed because `iso.a4` is `None`, but `a3` and `b3` apparently are not `None` (see https://travis-ci.org/astropy/photutils/jobs/621673519#L3958).

 In any case, I think your fix is clearer and more robust.